apps/site/lib/site_web/controllers/schedule/timetable_controller.ex

```diff
@@ -58,16 +58,65 @@ defmodule SiteWeb.ScheduleController.TimetableController do
       all_stops: all_stops
     } = build_timetable(conn.assigns.all_stops, timetable_schedules)
 
+    track_changes = track_changes(trip_schedules, route, direction_id)
+
     conn
     |> assign(:timetable_schedules, timetable_schedules)
     |> assign(:header_schedules, header_schedules)
     |> assign(:trip_schedules, trip_schedules)
+    |> assign(:track_changes, track_changes)
     |> assign(:vehicle_schedules, vehicle_schedules)
     |> assign(:prior_stops, prior_stops)
-    |> assign(:trip_messages, trip_messages(conn.assigns.route, conn.assigns.direction_id))
+    |> assign(:trip_messages, trip_messages(route, direction_id))
     |> assign(:all_stops, all_stops)
   end
 
+  @spec track_changes(
+          %{required({Schedules.Trip.id_t(), Stops.Stop.id_t()}) => Schedules.Schedule.t()},
+          Route.t(),
+          0 | 1
+        ) :: %{
+          required({Schedules.Trip.id_t(), Stops.Stop.id_t()}) => String.t() | nil
+        }
+  defp track_changes(trip_schedules, route, direction_id) do
+    Map.new(trip_schedules, fn {{trip_id, stop_id}, sch} ->
+      # Fetches all the predictions on the route in a direction, for a trip
+      predictions =
+        Predictions.Repo.all(
```

Comment:
question: Do we want this to use the non-cached version of predictions?  I am not sure the TTL of the cache is short enough for what this feature wants.